### PR TITLE
v1.7 backports 2020-06-30

### PR DESCRIPTION
**Pull request by nebril**
* #11231 -- Protect ENI and Azure IPAM from misbehaving cloud APIs (@tgraf)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 11231; do contrib/backporting/set-labels.py $pr done 1.7; done
```

### CODE REVIEW

**Comment by nebril** [comment]
test-backport-1.7

**Comment by nebril** [comment]
test-backport-1.7

**Comment by nebril** [comment]
test-missed-k8s

File: pkg/aws/eni/node_manager.go [hunk 1]
```diff
@@ -119,9 +119,34 @@ func NewNodeManager(instancesAPI nodeManagerAPI, ec2API ec2API, k8sAPI k8sAPI, m
...
+
+// SetInstancesAPIReadiness sets the readiness state of the instances API
+func (n *NodeManager) SetInstancesAPIReadiness(ready bool) {
+	n.mutex.Lock()
+	n.stableInstancesAPI = ready
+	n.mutex.Unlock()
+}
+
+// InstancesAPIIsReady returns true if the instances API is stable and ready
+func (n *NodeManager) InstancesAPIIsReady() bool {
```
Comment (joestringer) [inline pkg/aws/eni/node_manager.go:144]: Where did the user of this function go?

**Review by joestringer — CHANGES_REQUESTED** [review]
First two commits look good.

Third commit is a bit less straightforward to trace. I have one question below, would be nice if after your response we have someone else take a look over this PR as well.

File: pkg/aws/eni/node_manager.go (hunk 1 above)
Comment (nebril) [reply pkg/aws/eni/node_manager.go:144]: Great catch! Adding this to last commit.

**Comment by nebril** [comment]
test-backport-1.7

**Comment by nebril** [comment]
test-missed-k8s

**Comment by nebril** [comment]
@joestringer fixed, but I think it would be good to have @tgraf  take a look as original author

**Review by tgraf — APPROVED** [review]

**Pull request merged**